src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/BaseOptions.java

```diff
@@ -338,6 +352,34 @@ public boolean process(String opt, List<String> args) {
                     }
                 },
 
+                new XOption(resources, "--date", 1) {
+                    // Valid --date range: within a year of now
```

Comment:
Adding "--date" looks reasonable and makes it consistent with jar/jmod. I'm just not sure about restricting the date to "within a year of now". I assume projects will use the SOURCE_DATE_EPOCH to create a value for --date so they will run into issues if someone tries to build the project again, in a year.

Reply:
I'll change the range to 10 years. 

I was expecting that the common use of reproducible builds was to compare builds created within a reasonably short period of time. I was wanting to set a range that was less than the seemingly arbitrary range of 1980-2099 allowed by the jar tool.

Reply:
> I was expecting that the common use of reproducible builds was to compare builds created within a reasonably short period of time. I was wanting to set a range that was less than the seemingly arbitrary range of 1980-2099 allowed by the jar tool.

The restriction that the year be in the range 1980-2099 is because of the legacy ZIP format and the use of extended DOS time stamps. It would introducing new APIs to support a wider date range. I don't think javadoc needs to put any restriction on the range. I don't have any objection to 10 years, I was mostly curious why it is needed.

Reply:
I was following the jmod code, which uses a range, and I guessed (apparently correctly) that the chosen range was ip-specific.

I think it is worthwhile having a sanity-check on the provided range.